### PR TITLE
[Monitor Query] new migration guide

### PR DESCRIPTION
**Pull request by gracewilcox**
Adding migration guide to transition from `azquery` to `azlogs` and `azmetrics`

Part of https://github.com/Azure/azure-sdk-for-go/issues/22377

### CODE REVIEW

File: sdk/monitor/query/MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+}
+```
+
+<!-- LINKS -->
+[armmonitor]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/monitor/armmonitor
+[azlogs]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azlogs
+[azmetrics]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azmetrics
+[azmetrics_blog]: https://devblogs.microsoft.com/azure-sdk/multi-resource-metrics-query-support-in-the-azure-monitor-query-libraries/
+[github_issues]: https://github.com/Azure/azure-sdk-for-go/issues
+[rest_api]: https://learn.microsoft.com/en-us/rest/api/monitor/
```
Comment (scottaddie) [inline sdk/monitor/query/MIGRATION.md:206]: Omit the locale string

File: sdk/monitor/query/MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+}
+```
+
+### `armmonitor`
+
+The code in `armmonitor` is closer to the REST API; therefore, there are some name changes. Additionally, `azquery.MetricsClient` is separated into [different clients](#name-changes).
+
+```go
+import (
+    "context"
```
Comment (scottaddie) [inline sdk/monitor/query/MIGRATION.md:159]: Indentation on this line looks off

File: sdk/monitor/query/MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+Metrics has a larger split. Users wanting to use the existing ARM API's should use the [`armmonitor`][armmonitor] module. `armmonitor` contains all the metrics functionality from `azquery` and more.
+
+[`azmetrics`][azmetrics] contains the [new data plane metrics][azmetrics_blog] functionality. This new code was not previously generally available in `azquery`. The code allows users to query data from one or multiple resource IDs using a data plane endpoint. `azmetrics` will expand in the future when the service adds more features.
+
+### Name changes
+
+| old module   | old method name |new module | new method name | 
+| ----------- | ----------- | --- | --- |
+| `azquery` | LogsClient.QueryWorkspace | `azlogs` | Client.QueryWorkspace |
```
Comment (scottaddie) [inline sdk/monitor/query/MIGRATION.md:15]: Code-fence method names in this table

Comment (gracewilcox) [reply sdk/monitor/query/MIGRATION.md]: fixed with [c79390d](https://github.com/Azure/azure-sdk-for-go/pull/22753/commits/c79390ddba46a599c6717d0bde75b9d3ccb46996)

File: sdk/monitor/query/MIGRATION.md (hunk 2 above)
Comment (gracewilcox) [reply sdk/monitor/query/MIGRATION.md]: fixed with [c79390d](https://github.com/Azure/azure-sdk-for-go/pull/22753/commits/c79390ddba46a599c6717d0bde75b9d3ccb46996)

File: sdk/monitor/query/MIGRATION.md (hunk 1 above)
Comment (gracewilcox) [reply sdk/monitor/query/MIGRATION.md]: fixed with [c79390d](https://github.com/Azure/azure-sdk-for-go/pull/22753/commits/c79390ddba46a599c6717d0bde75b9d3ccb46996)

File: sdk/monitor/query/MIGRATION.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
+# Guide to migrate from `azquery` to `azlogs`, `azmetrics`, and `armmonitor`
+
+With the service team's creation of a new data plane metrics endpoint, the Go SDK team decided to reimagine our Monitor story. After careful deliberation, we decided to restructure Monitor Query to more accurately reflect the [REST API][rest_api].
+
+Logs largely remained the same; the code just moved from `azquery` to [`azlogs`][azlogs].
+
+Metrics has a larger split. Users wanting to use the existing ARM API's should use the [`armmonitor`][armmonitor] module. `armmonitor` contains all the metrics functionality from `azquery` and more.
+
+[`azmetrics`][azmetrics] contains the [new data plane metrics][azmetrics_blog] functionality. This new code was not previously generally available in `azquery`. The code allows users to query data from one or multiple resource IDs using a data plane endpoint. `azmetrics` will expand in the future when the service adds more features.
```
Comment (chlowell) [inline sdk/monitor/query/MIGRATION.md:9]:
```suggestion
[`azmetrics`][azmetrics] contains a [new data plane metrics feature][azmetrics_blog] that wasn't previously available in `azquery`. This allows users to query data from one or multiple resource IDs using a data plane endpoint. `azmetrics` will expand in the future when the service adds more features.
```

File: sdk/monitor/query/MIGRATION.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+| ----------- | ----------- | --- | --- |
+| `azquery` | LogsClient.QueryWorkspace | `azlogs` | Client.QueryWorkspace |
+|  | LogsClient.QueryResource |  | Client.QueryResource |
+| | LogsClient.QueryBatch | | N/A |
+| | MetricsClient.QueryResource | `armmonitor` | MetricsClient.List |
+| | MetricsClient.NewListDefinitionsPager  | | MetricDefinitionsClient.NewListPager |
+| | MetricsClient.NewListNamespacesPager  | | MetricNamespacesClient.NewListPager |
+|  | N/A | `azmetrics` | Client.QueryResources | 
+
+The `azlogs` module does not contain the `QueryBatch` method. To request that functionality, please [file an issue in our github repo][github_issues], so we can prioritize adding it to `azlogs`.
```
Comment (chlowell) [inline sdk/monitor/query/MIGRATION.md:23]: Is this something we already plan to do?

Comment (gracewilcox) [reply sdk/monitor/query/MIGRATION.md:23]: @scottaddie Is the service team still planning to deprecate Batch Query? Was not going to add it to the sdk at all if they are. 

If not, was thinking on waiting to see if there's any demand for batch. It's an ugly API.

File: sdk/monitor/query/MIGRATION.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+
+With the service team's creation of a new data plane metrics endpoint, the Go SDK team decided to reimagine our Monitor story. After careful deliberation, we decided to restructure Monitor Query to more accurately reflect the [REST API][rest_api].
+
+Logs largely remained the same; the code just moved from `azquery` to [`azlogs`][azlogs].
+
+Metrics has a larger split. Users wanting to use the existing ARM APIs should use the [`armmonitor`][armmonitor] module. `armmonitor` contains all the metrics functionality from `azquery` and more.
+
+[`azmetrics`][azmetrics] contains a [new data plane metrics feature][azmetrics_blog] that wasn't previously available in `azquery`. This allows users to query data from one or multiple resource IDs using a data plane endpoint. `azmetrics` will expand in the future when the service adds more features.
+
+### Name changes
```
Comment (scottaddie) [inline sdk/monitor/query/MIGRATION.md:11]:
```suggestion
## Name changes
```

File: sdk/monitor/query/MIGRATION.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+| old module   | old method name |new module | new method name |
```
Comment (scottaddie) [inline sdk/monitor/query/MIGRATION.md:13]:
```suggestion
| Old module   | Old method name |New module | New method name | 
```

Comment (gracewilcox) [reply sdk/monitor/query/MIGRATION.md]: fixed in [b994525](https://github.com/Azure/azure-sdk-for-go/pull/22753/commits/b994525bed1636c83155a3932ce00aeb00e2fc70)

File: sdk/monitor/query/MIGRATION.md (hunk 6 above)
Comment (gracewilcox) [reply sdk/monitor/query/MIGRATION.md]: fixed in [b994525](https://github.com/Azure/azure-sdk-for-go/pull/22753/commits/b994525bed1636c83155a3932ce00aeb00e2fc70)

File: sdk/monitor/query/MIGRATION.md (hunk 5 above)
Comment (gracewilcox) [reply sdk/monitor/query/MIGRATION.md]: fixed in [b994525](https://github.com/Azure/azure-sdk-for-go/pull/22753/commits/b994525bed1636c83155a3932ce00aeb00e2fc70)

removing paragraph

File: sdk/monitor/query/MIGRATION.md [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+
+| Old module   | Old method name |New module | New method name | 
+| ----------- | ----------- | --- | --- |
+| `azquery` | `LogsClient.QueryWorkspace` | `azlogs` | `Client.QueryWorkspace` |
+|  | `LogsClient.QueryResource` |  | `Client.QueryResource` |
+| | `LogsClient.QueryBatch` | | N/A |
+| | `MetricsClient.QueryResource` | `armmonitor` | `MetricsClient.List` |
+| | `MetricsClient.NewListDefinitionsPager`  | | `MetricDefinitionsClient.NewListPager` |
+| | `MetricsClient.NewListNamespacesPager`  | | `MetricNamespacesClient.NewListPager` |
+|  | N/A | `azmetrics` | `Client.QueryResources` |
```
Comment (chlowell) [inline sdk/monitor/query/MIGRATION.md:21]: If this is an entirely new method with no equivalent in the old module, I wouldn't call it a name change
```suggestion
```

File: sdk/monitor/query/MIGRATION.md [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+[azlogs]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azlogs
+[azmetrics]: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/monitor/query/azmetrics
+[azmetrics_blog]: https://devblogs.microsoft.com/azure-sdk/multi-resource-metrics-query-support-in-the-azure-monitor-query-libraries/
+[github_issues]: https://github.com/Azure/azure-sdk-for-go/issues
+[rest_api]: https://learn.microsoft.com/rest/api/monitor/
+
+
+
+
+
```
[the remark this answers is not in the document]
Comment (gracewilcox) [reply sdk/monitor/query/MIGRATION.md]: fixed in [05248e7](https://github.com/Azure/azure-sdk-for-go/pull/22753/commits/05248e7745d328748da47cc44a9df680b9d6584e)

File: sdk/monitor/query/MIGRATION.md (hunk 8 above)
Comment (gracewilcox) [reply sdk/monitor/query/MIGRATION.md]: fixed in [05248e7](https://github.com/Azure/azure-sdk-for-go/pull/22753/commits/05248e7745d328748da47cc44a9df680b9d6584e)

**Pull request merged**